### DISCUSSION TITLE
Welcome to doatask Discussions!

### DISCUSSION THREAD

**Discussion by tk1** [tk1/doatask#16] [Ideas]
# XXXCodeRunner

Wäre es nicht eine Möglichkeit, statt in jeder Sprache einen Webserver zu implementieren, einen einzigen in Node geschriebenen Webserver zu haben, der die Kommunikation mit außen übernimmt und im Container das auszuführende Programm startet und dessen Ergebnis zurückliefert? 

Dann könnte man z.B. im JavaCodeRunner einfach den Java-Compiler aufrufen und dann das kompilierte Programm.

**Comment by tglaes** [comment]
Ja das ist richtig. Dann muss man im Container mit Dateien arbeiten und immer noch node zusätzlich installieren. Dadurch werden die Container größer und und die Implementierung ist etwas komplexer. Aber generell eine gute Idee.

**Comment by tk1** [comment]
Wie Sie schon bemerkt haben, schreiben wir hier auf Deutsch. Englisch ist nur dann zwingend erforderlich, wenn es im Repository landet, also Code, Kommentare im Code, Commit-Beschreibungen etc.
